spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,161 @@
+          create_list(:book, 5, genre: 'Fantasy')
+          get :index, params: { genre: 'Action' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+
+        it 'responses with a json with 5 books' do
+          expect(response_body['page'].length).to eq 5
+        end
+        it 'returns only books with genre == Action' do
+          expect(response_body['page'].map { |book| book['genre'] }).to satisfy do |genres|
+            genres.all? { |genre| genre == 'Action' }
+          end
+        end
+      end
+
+      context 'When params[:author] == Bradford Nolan DDS' do
+        before do
+          create_list(:book, 2, author: 'Bradford Nolan DDS')
+          create_list(:book, 1, author: 'Luis Miguel')
+          get :index, params: { author: 'Bradford Nolan DDS' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+        it 'responses with a json with 2 books' do
+          expect(response_body['page'].length).to eq 2
+        end
+        it 'returns only books with author == Bradford Nolan DDS' do
+          expect(response_body['page'].map { |book| book['author'] }).to satisfy do |authors|
+            authors.all? { |author| author == 'Bradford Nolan DDS' }
+          end
+        end
+      end
+
+      context 'When params[:author] == Bradford Nolan DDS' do
+        before do
+          create_list(:book, 2, author: 'Bradford Nolan DDS')
+          create_list(:book, 1, author: 'Luis Miguel')
+          get :index, params: { author: 'Bradford Nolan DDS' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+        it 'responses with a json with 2 books' do
+          expect(response_body['page'].length).to eq 2
+        end
+        it 'returns only books with author == Bradford Nolan DDS' do
+          expect(response_body['page'].map { |book| book['author'] }).to satisfy do |authors|
+            authors.all? { |author| author == 'Bradford Nolan DDS' }
+          end
+        end
+      end
+
+      context 'When params[:title] == La biblia!' do
+        before do
+          create(:book, title: 'La biblia!')
+          create(:book, title: 'Harry Potter')
+          get :index, params: { title: 'La biblia!' }
+        end
+        subject { response_body }
+
+        it { is_expected.to be_paginated }
+        it 'responses with a json with 1 book' do
+          expect(response_body['page'].length).to eq 1
+        end
+        it 'returns only a book with title == La biblia!' do
+          expect(response_body['page'][0]['title']).to eq 'La biblia!'
+        end
+      end
+    end
+    context 'When there is not an authenticated user' do
+      context 'When fetching all the books' do
+        before { get :index }
+        it { is_expected.to respond_with :unauthorized }
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    let(:book) { create(:book) }
```

Comment:
We are going to create books in contexts :wink: 
```suggestion
    subject(:http_request) { get :show, params: { id: book.id }  }
```